apps/src/lib/client/eth_bridge.rs

```diff
@@ -1,2 +1,30 @@
 pub mod bridge_pool;
 pub mod validator_set;
+
+use std::ops::ControlFlow;
+
+use tokio::time::{Duration, Instant};
+use web30::client::Web3;
+
+use crate::node::ledger::ethereum_oracle::eth_syncing_status;
+use crate::timeouts::TimeoutStrategy;
+
+/// Block until Ethereum finishes synchronizing.
+pub async fn block_on_eth_sync(deadline: Instant, url: &str) {
```

Comment:
Is this really the strategy we want if ethereum is not synced? If it is far behind, this method will take a long to time to resolve, hijacking the cli, forcing the user to kill it. Maybe it is just better to stop the operation with a long and maybe some helpful info about when to try again.